src/log/Log.cc

```diff
@@ -372,7 +372,13 @@ void Log::dump_recent()
   {
     char pthread_name[16] = {0}; //limited by 16B include terminating null byte.
     ceph_pthread_getname(pthread_id, pthread_name, sizeof(pthread_name));
-    _log_message(fmt::format("  {} / {}", pthread_id, pthread_name), true);
+    uint64_t tid;
+    if constexpr (std::is_pointer_v<decltype(pthread_id)>) {
+      tid = reinterpret_cast<std::uintptr_t>(pthread_id);
+    } else {
+      tid = pthread_id;
```

Comment:
```
/home/jenkins/workspace/ceph-master-compile/src/log/Log.cc:379:13: error: incompatible pointer to integer conversion assigning to 'uint64_t' (aka 'unsigned long') from 'pthread *const'
      tid = pthread_id;
            ^~~~~~~~~~
1 error generated.

```

Reply:
@wjwithagen we need to templatize this method so the compiler can skip the unused branch. fixed and repushed.